Content.Server/Nyanotrasen/StationEvents/Events/GlimmerShadeSpawnRule.cs

```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using Robust.Shared.Random;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Shared.Atmos.Miasma;
+using Content.Server.StationEvents.Components;
+using Content.Shared.Psionics.Glimmer;
+
+namespace Content.Server.StationEvents.Events;
+
+internal sealed class GlimmerShadeSpawnRule : StationEventSystem<GlimmerShadeRuleComponent>
+{
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly GlimmerSystem _glimmerSystem = default!;
+
+    private static readonly string ShadePrototype = "MobShade";
+
+    protected override void Started(EntityUid uid, GlimmerShadeRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args);
+
+        var rottingPlaces = EntityManager.EntityQuery<RottingComponent, TransformComponent>().ToList();
```

Comment:
No need to convert and store this as a list if it's just being iterated over. `EntityQuery` should work fine here too.